src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -1,589 +1,131 @@
 
-import cloud.eppo.ShardUtils;
-import cloud.eppo.Utils;
-import cloud.eppo.rac.Constants;
-import cloud.eppo.rac.dto.*;
-import cloud.eppo.rac.exception.EppoClientIsNotInitializedException;
-import cloud.eppo.rac.exception.InvalidInputException;
-import com.eppo.sdk.helpers.*;
-import com.eppo.sdk.helpers.bandit.BanditEvaluator;
-import com.fasterxml.jackson.databind.JsonNode;
-import java.util.*;
-import java.util.stream.Collectors;
-import org.ehcache.Cache;
+import cloud.eppo.BaseEppoClient;
+import cloud.eppo.logging.AssignmentLogger;
+import cloud.eppo.logging.BanditLogger;
+import com.eppo.sdk.helpers.AppDetails;
+import com.eppo.sdk.helpers.FetchConfigurationsTask;
+import java.util.Timer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class EppoClient {
+public class EppoClient extends BaseEppoClient {
   private static final Logger log = LoggerFactory.getLogger(EppoClient.class);
-  private static EppoClient instance = null;
-  private final ConfigurationStore configurationStore;
-  private final Timer poller;
-  private final EppoClientConfig eppoClientConfig;
 
-  private EppoClient(
-      ConfigurationStore configurationStore, Timer poller, EppoClientConfig eppoClientConfig) {
-    this.configurationStore = configurationStore;
-    this.poller = poller;
-    this.eppoClientConfig = eppoClientConfig;
-  }
-
-  /** This function is used to get assignment Value */
-  protected Optional<EppoValue> getAssignmentValue(
-      String subjectKey,
-      String flagKey,
-      EppoAttributes subjectAttributes,
-      Map<String, EppoAttributes> actionsWithAttributes) {
-    Optional<Variation> assignedVariation =
-        getAssignmentVariation(subjectKey, flagKey, subjectAttributes, actionsWithAttributes);
-    return assignedVariation.map(Variation::getTypedValue);
-  }
-
-  /** Returns the assigned variation. */
-  public Optional<Variation> getAssignmentVariation(
-      String subjectKey, String flagKey, EppoAttributes subjectAttributes) {
-    return getAssignmentVariation(subjectKey, flagKey, subjectAttributes, null);
-  }
+  private static final String DEFAULT_HOST = "https://fscdn.eppo.cloud";
+  private static final boolean DEFAULT_IS_GRACEFUL_MODE = true;
+  private static final boolean DEFAULT_FORCE_REINITIALIZE = false;
+  private static final long DEFAULT_POLLING_INTERVAL_MS = 30 * 1000;
+  private static final long DEFAULT_JITTER_INTERVAL_RATIO = 10;
 
-  /** Returns the assigned variation. */
-  protected Optional<Variation> getAssignmentVariation(
-      String subjectKey,
-      String flagKey,
-      EppoAttributes subjectAttributes,
-      Map<String, EppoAttributes> actionsWithAttributes) {
-    // Validate Input Values
-    InputValidator.validateNotBlank(subjectKey, "Invalid argument: subjectKey cannot be blank");
-    InputValidator.validateNotBlank(flagKey, "Invalid argument: flagKey cannot be blank");
+  private static EppoClient instance;
+  private static Timer pollTimer;
```

Comment:
The poller made static so that it is a singleton as well, and we don't have renegade timers everywhere. Also so it can easily be stopped (useful for tests)